spec/factories/dialog.rb

```diff
@@ -1,4 +1,6 @@
 FactoryGirl.define do
   factory :dialog do
+    # skip validate_children callback for general dialog testing
+    before(:create) { |d| d.stub(:validate_children) }
```

Comment:
Skip checking the children upon FactoryGirl.create(:dialog, ....) which is called in lots of test cases where the validation of children is irrelevant.